### PR TITLE
Implement batcher CLI As Short Term

### PR DESCRIPTION
**Pull request by longquanzheng**
We will re-implement it with service API after https://github.com/uber/cadence/pull/2220 is released.

Tested locally:
```
longer@~/gocode/src/github.com/uber/cadence:(batch-cli)$ ./cadence --do samples-domain wf batch start --query "WorkflowType='main.SampleParentWorkflow'" --reason "test" --bt signal --sig testname --input haha
This batch job will be operating on 5 workflows.
Please confirm[Yes/No]:yes
{
  "jobID": "8e131345-60a3-4bbc-b2a8-37d81d6f5f02",
  "msg": "batch job is started"
}

longer@~/gocode/src/github.com/uber/cadence:(batch-cli)$ ./cadence --do samples-domain wf batch list -ps 2
[
  {
    "closeTime": "2019-07-18T15:14:06-07:00",
    "jobID": "8e131345-60a3-4bbc-b2a8-37d81d6f5f02",
    "operator": "\"longer\"",
    "reason": "\"test\"\n",
    "startTime": "2019-07-18T15:14:06-07:00",
    "status": "COMPLETED"
  },
  {
    "closeTime": "2019-07-18T15:11:46-07:00",
    "jobID": "5e9d0692-3f57-4dac-9b2f-1d68f3fedf9e",
    "operator": "\"longer\"",
    "reason": "\"test\"\n",
    "startTime": "2019-07-18T15:11:45-07:00",
    "status": "COMPLETED"
  }
]

longer@~/gocode/src/github.com/uber/cadence:(batch-cli)$ ./cadence --do samples-domain wf batch desc -jid 8e131345-60a3-4bbc-b2a8-37d81d6f5f02
{
  "msg": "batch job is finished successfully"
}


```

### CODE REVIEW

File: tools/cli/workflow.go [hunk 1]
```diff
@@ -364,3 +371,105 @@ func newActivityCommands() []cli.Command {
...
+func newBatchCommands() []cli.Command {
+	return []cli.Command{
+		{
+			Name:    "describe",
+			Aliases: []string{"desc"},
+			Usage:   "Describe a batch operation job",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  FlagWorkflowIDWithAlias,
+					Usage: "WorkflowID",
```
Comment (mfateev) [inline tools/cli/workflow.go]: Why is it WorkflowID not jobID?

File: tools/cli/workflow.go [hunk 2]
```diff
@@ -364,3 +371,105 @@ func newActivityCommands() []cli.Command {
...
+				},
+				cli.IntFlag{
+					Name:  FlagRPS,
+					Value: batcher.DefaultRPS,
+					Usage: "RPS of processing",
+				},
+				cli.IntFlag{
+					Name:  FlagConcurrency,
+					Value: batcher.DefaultConcurrency,
+					Usage: "Number of goroutines of concurrent processing",
```
Comment (mfateev) [inline tools/cli/workflow.go]: Why are these exposed? Shouldn't this be internal service configuration?

File: tools/cli/workflow.go [hunk 3]
```diff
@@ -364,3 +371,105 @@ func newActivityCommands() []cli.Command {
...
+				},
+				cli.IntFlag{
+					Name:  FlagAttemptsOnRetryableError,
+					Value: batcher.DefaultAttemptsOnRetryableError,
+					Usage: "Number of attempts on retryable error before giving up",
```
Comment (mfateev) [inline tools/cli/workflow.go]: I think this should be also hidden

File: tools/cli/workflow.go [hunk 4]
```diff
@@ -364,3 +371,105 @@ func newActivityCommands() []cli.Command {
...
+				},
+				cli.StringSliceFlag{
+					Name:  FlagNonRetryableError,
+					Usage: "Errors that are not retryable",
```
Comment (mfateev) [inline tools/cli/workflow.go]: What is the use case for these?

File: tools/cli/workflow.go [hunk 5]
```diff
@@ -364,3 +371,105 @@ func newActivityCommands() []cli.Command {
...
+				},
+				cli.DurationFlag{
+					Name:  FlagHeartbeatTimeout,
+					Value: batcher.DefaultActivityHeartBeatTimeout,
+					Usage: "heartbeat timeout for the system workflow activity",
+				},
```
Comment (mfateev) [inline tools/cli/workflow.go]: Should be hidden

File: tools/cli/workflow.go [hunk 6]
```diff
@@ -364,3 +371,105 @@ func newActivityCommands() []cli.Command {
...
+				cli.DurationFlag{
+					Name:  FlagExecutionTimeout,
+					Value: 6 * time.Hour,
+					Usage: "how long the batch job should run",
```
Comment (mfateev) [inline tools/cli/workflow.go]: Should be hidden

File: tools/cli/workflow.go [hunk 7]
```diff
@@ -364,3 +371,105 @@ func newActivityCommands() []cli.Command {
...
+				},
+			},
+			Action: func(c *cli.Context) {
+				DescribeBatchJob(c)
+			},
+		},
+		{
+			Name:    "list",
+			Aliases: []string{"l"},
+			Usage:   "Describe a batch operation job",
```
Comment (vancexu) [inline tools/cli/workflow.go]: This need update

File: tools/cli/workflow.go (hunk 1 above)
Comment (longquanzheng) [reply tools/cli/workflow.go]: changed to jobID

File: tools/cli/workflow.go (hunk 2 above)
Comment (longquanzheng) [reply tools/cli/workflow.go]: Deleted

File: tools/cli/workflow.go (hunk 3 above)
Comment (longquanzheng) [reply tools/cli/workflow.go]: Deleted

File: tools/cli/workflow.go (hunk 4 above)
Comment (longquanzheng) [reply tools/cli/workflow.go]: Deleted.

File: tools/cli/workflow.go (hunk 5 above)
Comment (longquanzheng) [reply tools/cli/workflow.go]: Deleted

File: tools/cli/workflow.go (hunk 6 above)
Comment (longquanzheng) [reply tools/cli/workflow.go]: Deleted. Batch job will run forever if it needs to.

File: tools/cli/workflow.go [hunk 8]
```diff
@@ -364,3 +370,89 @@ func newActivityCommands() []cli.Command {
...
+					Name:  FlagJobIDWithAlias,
+					Usage: "Batch Job ID",
+				},
+			},
+			Action: func(c *cli.Context) {
+				DescribeBatchJob(c)
+			},
+		},
+		{
+			Name:  "stop",
```
Comment (mfateev) [inline tools/cli/workflow.go]: s/stop/terminate to be consistent with workflows

File: tools/cli/workflow.go [hunk 9]
```diff
@@ -364,3 +370,89 @@ func newActivityCommands() []cli.Command {
...
+					Usage: "Result page size",
+				},
+			},
+			Action: func(c *cli.Context) {
+				ListBatchJobs(c)
+			},
+		},
+		{
+			Name:  "start",
+			Usage: "Start a batch operation job",
```
Comment (mfateev) [inline tools/cli/workflow.go]: I would add confirmation prompt that says something like:
You are about to terminate 100000 workflows. Are you sure?
n/y
and a flag to disable the confirmation

File: tools/cli/workflow.go (hunk 8 above)
Comment (longquanzheng) [reply tools/cli/workflow.go]: Done

File: tools/cli/workflow.go (hunk 9 above)
Comment (longquanzheng) [reply tools/cli/workflow.go]: Done

File: tools/cli/workflow.go [hunk 10]
```diff
@@ -449,6 +449,10 @@ func newBatchCommands() []cli.Command {
 					Value: batcher.DefaultRPS,
 					Usage: "RPS of processing",
 				},
+				cli.BoolFlag{
+					Name:  FlagYes,
+					Usage: "Optional flag to kip entering yes for confirmation",
```
Comment (mfateev) [inline tools/cli/workflow.go]: Disables confirmation prompt

File: tools/cli/workflowBatchCommands.go [hunk 11]
```diff
@@ -137,6 +139,34 @@ func StartBatchJob(c *cli.Context) {
...
+			text, err := reader.ReadString('\n')
+			if err != nil {
+				ErrorAndExit("Failed to  get confirmation for starting a batch job", err)
+			}
+			if strings.EqualFold(text, "no") {
+				ErrorAndExit("Batch job is not started", nil)
+			} else if text == "yes" {
+				break
+			} else {
+				fmt.Printf("not supported input %v\n", text)
```
Comment (mfateev) [inline tools/cli/workflowBatchCommands.go]: Change it to default to No as most CLIs do.

**Pull request merged**